### PR TITLE
DS-4123: ITs to prove access restricted Items cannot be accessed anonymously

### PR DESCRIPTION
**Pull request by Micheleboychuk**
See https://jira.lyrasis.org/browse/DS-4123

The most important change is in `dspace-server-webapp/src/main/java/org/dspace/app/rest/security/AuthorizeServicePermissionEvaluatorPlugin.java` line 88 that has been switched from exact check of the permission to the default inherit mechanism of DSpace. Please note that use of the inherit mechanism is the default in dspace so I would prefer to deal here in future with potential exception instead than make the PreAuthorization annotation more complicated/custom for normal scenario (again as inherit is the default).
For instance the existing test in `dspace-server-webapp/src/test/java/org/dspace/app/rest/ItemOwningCollectionUpdateRestControllerIT.java` `moveItemTestByAuthorizedUserWithoutWrite` was assuming a forbidden response now fixed in OK. Indeed, in the same conditions the current version allows the operation (tested on demo.dspace.org)

### CODE REVIEW

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/ItemRestRepositoryIT.java [hunk 1]
```diff
@@ -1294,6 +1295,102 @@ public void embargoAccessTest() throws Exception {
...
+        // collection1's admin user is allowed access to embargoed item
+        String tokenAdminCollection1 = getAuthToken(adminCollection1.getEmail(), "qwerty02");
+        getClient(tokenAdminCollection1).perform(get("/api/core/items/" + embargoedItem.getID()))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", is(ItemMatcher.matchItemProperties(embargoedItem))));
+
+        // collection2's admin user is NOT allowed access to embargoed item of collection1
+        String tokenAdminCollection2 = getAuthToken(adminCollection2.getEmail(), "qwerty03");
+        getClient(tokenAdminCollection2).perform(get("/api/core/items/" + embargoedItem.getID()))
+                .andExpect(status().isForbidden());
```
Comment (tdonohue) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/ItemRestRepositoryIT.java]: Two other tests are necessary in this method:
1. We need a test that a Full Admin can see the Embargoed Item.
2. Can we create a second Community which is *not* a parent of `collection1` and ensure that an Admin of that new Community is also Forbidden?

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/ItemRestRepositoryIT.java [hunk 2]
```diff
@@ -1422,6 +1519,118 @@ public void privateGroupAccessTest() throws Exception {
...
+        //A member of the restricted group is also allowed access to restricted item
+        String tokenMemberRestrictedGroup = getAuthToken(memberRestrictGroup.getEmail(), "qwerty01");
+        getClient(tokenMemberRestrictedGroup).perform(get("/api/core/items/" + itemRestrictedByGroup.getID()))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", is(ItemMatcher.matchItemProperties(itemRestrictedByGroup))));
+
+        //members who are not part of the restricted group, have no access to the item
+        String tokenEPerson = getAuthToken(eperson.getEmail(), password);
+        getClient(tokenEPerson).perform(get("/api/core/items/" + itemRestrictedByGroup.getID()))
+                .andExpect(status().isForbidden());
```
Comment (tdonohue) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/ItemRestRepositoryIT.java]: We should also have a test here to verify that anonymous users are Forbidden

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/ItemRestRepositoryIT.java [hunk 3]
```diff
@@ -1422,6 +1519,118 @@ public void privateGroupAccessTest() throws Exception {
...
+        // collection1's admin user is allowed access to restricted item
+        String tokenAdminCollection1 = getAuthToken(adminCollection1.getEmail(), "qwerty02");
+        getClient(tokenAdminCollection1).perform(get("/api/core/items/" + itemRestrictedByGroup.getID()))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", is(ItemMatcher.matchItemProperties(itemRestrictedByGroup))));
+
+        // collection2's admin user is NOT allowed access to restricted item of collection1
+        String tokenAdminCollection2 = getAuthToken(adminCollection2.getEmail(), "qwerty03");
+        getClient(tokenAdminCollection2).perform(get("/api/core/items/" + itemRestrictedByGroup.getID()))
+                .andExpect(status().isForbidden());
```
Comment (tdonohue) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/ItemRestRepositoryIT.java]: Again, here it'd be good to create another Community that is *not* a parent of collection1, and ensure that a Community Admin of that new Community is also Forbidden.

**Review by tdonohue — COMMENTED** [review]
@Micheleboychuk : I gave this a review today. It looks good overall, but there a few more tests that I'd like to see added here before we can merge it.  See comments inline below.

**Comment by abollini** [comment]
@tdonohue this should be ready to go now

**Review by tdonohue — APPROVED** [review]
👍 This looks good to me now. Thanks @Micheleboychuk and @abollini .  Merging this as it's almost entirely ITs and was flagged for one approval by the DSpace 7 WG.

**Pull request merged**